src/main.rs

```diff
@@ -163,35 +138,69 @@ impl BFInterp {
             MetadataOperand => todo!(),
         };
 
-        let mut string_temp;
-        for arg in call.arguments[1..].iter() {
-            match &arg.0 {
-                LocalOperand { name, .. } => {
-                    string_temp =
-                        string.replacen("{}", self.mem.get(&name).unwrap().to_string().as_str(), 1);
-                    string = &string_temp;
-                }
-                ConstantOperand(con_op) => match con_op.as_ref() {
-                    Constant::Int { value, .. } => {
-                        string_temp = string.replacen("{}", value.to_string().as_str(), 1);
-                        string = &string_temp;
+        let mut string = String::new();
+        let mut arg_it = call.arguments[1..].into_iter();
+        match constant {
+            Constant::Array { elements, .. } => {
+                let mut i = 0;
+                while i < elements.len() {
+                    match elements[i].as_ref() {
+                        Constant::Int { value, .. } => {
+                            let char = *value as u8 as char;
```

Comment:
`char` is a reserved word since it's a type. I'm actually surprised Rust still lets you use that as a variable name. In any case, it's confusing to use reserved words as variables especially if you use both in the same line, so maybe rename this to `ch`.